content/components/avoid-functions-in-templates.md

```diff
@@ -0,0 +1,101 @@
+---
+title: avoid functions in templates
+---
+
+# Problem
+
+Since angular can not know if result of the function (or a getter) has changed, in default ChangeDetection mode the component will continuously execute template bound function to keep up-to-date with underlying data flow. 
+
+This can cause performance issues and hide template complexity.
+
+# Solution #1
+
+Functions that will always result the same value during the lifetime of the component execute on `ngOnInit` or even earlier in `constructor` and assign to component public property which will then be used in its template.
+
+
+```ts
+@Component({
+  template: `<div *ngIf="shouldShowContent()">content to show</div>`
+})
+export class SomeComponent {
+  users: User[];
+  visible: boolean;
+
+  constructor(private userService: UserService, private deviceService: DeviceService){
+  }
+
+  shouldShowContent(): boolean {
+    return this.userService.isUserAmazing() && this.deviceService.isUsingModernBrowser()
+  }
+
+}
+```
+
+## Becomes: 
+
+```ts
+@Component({
+  template: `<div *ngIf="shouldShowContent">content to show</div>`
+})
+export class SomeComponent {
+  users: User[];
+  visible: boolean;
+  shouldShowContent: boolean;
+
+  constructor(private userService: UserService, private deviceService: DeviceService){
+    this.shouldShowContent = this.getShouldShowContent();
```

Comment:
We try to advice people to NOT do anything in the constructor so I'd remove this part.